src/Infrastructure/Service/ValueReceiver/ApiInteractionProcessor.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\Sdk\Infrastructure\Service\ValueReceiver;
+
+use SprykerSdk\Sdk\Core\Application\Dependency\InteractionProcessorInterface;
+use SprykerSdk\Sdk\Core\Application\Dto\ReceiverValueInterface;
+use SprykerSdk\Sdk\Infrastructure\Event\RequestDataInjectorInterface;
+use SprykerSdk\Sdk\Infrastructure\Exception\InvalidRequestDataException;
+use SprykerSdk\SdkContracts\Enum\ValueTypeEnum;
+
+class ApiInteractionProcessor implements InteractionProcessorInterface, RequestDataInjectorInterface
+{
+    /**
+     * @var array
+     */
+    protected array $data;
+
+    /**
+     * @param array $data
+     *
+     * @return void
+     */
+    public function setRequestData(array $data): void
+    {
+        $this->data = $data;
+    }
+
+    /**
+     * @param string $key
+     *
+     * @return bool
+     */
+    public function has(string $key): bool
+    {
+        return !empty($this->data[$key]);
+    }
+
+    /**
+     * @param string $key
+     *
+     * @return mixed
+     */
+    public function get(string $key)
+    {
+        return $this->data[$key];
+    }
+
+    /**
+     * @param \SprykerSdk\Sdk\Core\Application\Dto\ReceiverValueInterface $receiverValue
+     *
+     * @throws \SprykerSdk\Sdk\Infrastructure\Exception\InvalidRequestDataException
+     *
+     * @return mixed
+     */
+    public function receiveValue(ReceiverValueInterface $receiverValue)
+    {
+        $choiceValues = $receiverValue->getChoiceValues() ? $this->prepareChoiceValues($receiverValue->getChoiceValues()) : [];
```

Comment:
`$choiceValues = $this->prepareChoiceValues($receiverValue->getChoiceValues());`